EFAuditer/EFAuditServiceExtensions.cs

```diff
@@ -26,7 +37,7 @@ public static void AddEFAuditer(this IServiceCollection services, string connect
                         audit.EventType = entry.Action;
                         audit.AuditDetails = ev.CustomFields.ContainsKey(CustomFields.AuditDetails) ? ev.CustomFields[CustomFields.AuditDetails].ToString() : null;
                         audit.AuditDateTime = DateTime.Now;
-                        audit.AuditUser = ev.Environment.UserName;
+                        audit.AuditUser = ev.CustomFields.ContainsKey(CustomFields.AppUser) ? ev.CustomFields[CustomFields.AppUser].ToString() : ev.Environment.UserName;

```

Comment:
Now we're doing this logic twice (same for AuditDetails), might it be worth extracting to a private method `private string GetCustomKey(CustomFields field)` and then using the null coalescing operator here
```
audit.AuditUser = GetCustomKey(CustomFields.AppUser) ?? ev.Environment.UserName
```

Reply:
good call!